### ISSUE TITLE
Cannot import wallet to Sparrow

### ISSUE THREAD

**Issue by bry-med** [sparrowwallet/sparrow#1308]
Receiving the following error: 

"Error opening wallet"
Cannot invoke "com.sparrowwallet.drongo.policy.Policy.getName()" because "< parameter1 >" is null

Moving from Specter as it started giving me issues after recent update. I have two exported JSON files and I was able to import one of them with no issues, but the older wallet (with approximately 10x the amount of transactions) gives the error above when I attempt to import.

Any help is much appreciated.

**Comment by craigraw** [comment]
My first suggestion is to simply import the wallet into Sparrow using one of the normal methods. If you have a hardware wallet, import it by creating a new wallet in Sparrow and following the standard process of importing the xpub using the Connected Hardware Wallet or Airgapped Hardware Wallet options. 

If you want to try the Specter Desktop import route again, please provide the full error from the log file. Please also take a look [at the example file here](https://github.com/sparrowwallet/sparrow/blob/d1ac5b076e7213610abab29b77a8d6fd1570bd0e/src/test/resources/com/sparrowwallet/sparrow/io/specter-wallet.json), and compare to your own in a text editor to see if there are any significant differences.

**Comment by bry-med** [comment]
Thanks for the response. When I upload my generic JSON from my Coldcard, I'm only seeing the transactions from one wallet (the one that I was able to import). Essentially I have two wallets that I created from my Coldcard--one for KYC bitcoin and one for non-KYC bitcoin. The non-KYC wallet is the one that I've been able to import without issues, but I can't import my KYC wallet. Both wallets were generated by my Coldcard/Specter setup.

When I compare my wallet file to the sample you provided, mine has a lot more information in it... "name", "description", "alias", "address_type", "address_index", etc.

Here is the full error log from Sparrow:

2024-03-09 10:27:08,469 ERROR [JavaFX Application Thread] c.s.s.AppController [null:-1] Error opening wallet
java.lang.NullPointerException: Cannot invoke "com.sparrowwallet.drongo.policy.Policy.getName()" because "<parameter1>" is null
	at com.sparrowwallet.sparrow@1.8.4/com.sparrowwallet.sparrow.io.db.PolicyDao.addPolicy(Unknown Source)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(Unknown Source)
	at com.sparrowwallet.merged.module@1.8.4/org.jdbi.v3.core.internal.exceptions.Unchecked.lambda$function$4(Unknown Source)
	at com.sparrowwallet.merged.module@1.8.4/org.jdbi.v3.sqlobject.DefaultMethodHandler.invoke(Unknown Source)
	at com.sparrowwallet.merged.module@1.8.4/org.jdbi.v3.sqlobject.internal.SqlObjectInitData$1.lambda$invoke$0(Unknown Source)
	at com.sparrowwallet.merged.module@1.8.4/org.jdbi.v3.core.internal.Invocations.invokeWith(Unknown Source)
	at com.sparrowwallet.merged.module@1.8.4/org.jdbi.v3.core.ConstantHandleSupplier.lambda$invokeInContext$0(Unknown Source)
	at com.sparrowwallet.merged.module@1.8.4/org.jdbi.v3.core.internal.Invocations.invokeWith(Unknown Source)
	at com.sparrowwallet.merged.module@1.8.4/org.jdbi.v3.core.ConstantHandleSupplier.invokeInContext(Unknown Source)
	at com.sparrowwallet.merged.module@1.8.4/org.jdbi.v3.sqlobject.internal.SqlObjectInitData$1.call(Unknown Source)
	at com.sparrowwallet.merged.module@1.8.4/org.jdbi.v3.sqlobject.internal.SqlObjectInitData$1.invoke(Unknown Source)
	at com.sparrowwallet.merged.module@1.8.4/org.jdbi.v3.sqlobject.SqlObjectFactory.lambda$attach$2(Unknown Source)
	at jdk.proxy2/jdk.proxy2.$Proxy30.addPolicy(Unknown Source)
	at com.sparrowwallet.sparrow@1.8.4/com.sparrowwallet.sparrow.io.db.WalletDao.addWallet(Unknown Source)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(Unknown Source)
	at com.sparrowwallet.merged.module@1.8.4/org.jdbi.v3.core.internal.exceptions.Unchecked.lambda$function$4(Unknown Source)
	at com.sparrowwallet.merged.module@1.8.4/org.jdbi.v3.sqlobject.DefaultMethodHandler.invoke(Unknown Source)
	at com.sparrowwallet.merged.module@1.8.4/org.jdbi.v3.sqlobject.internal.SqlObjectInitData$1.lambda$invoke$0(Unknown Source)
	at com.sparrowwallet.merged.module@1.8.4/org.jdbi.v3.core.internal.Invocations.invokeWith(Unknown Source)
	at com.sparrowwallet.merged.module@1.8.4/org.jdbi.v3.core.ConstantHandleSupplier.lambda$invokeInContext$0(Unknown Source)
	at com.sparrowwallet.merged.module@1.8.4/org.jdbi.v3.core.internal.Invocations.invokeWith(Unknown Source)
	at com.sparrowwallet.merged.module@1.8.4/org.jdbi.v3.core.ConstantHandleSupplier.invokeInContext(Unknown Source)
	at com.sparrowwallet.merged.module@1.8.4/org.jdbi.v3.sqlobject.internal.SqlObjectInitData$1.call(Unknown Source)
	at com.sparrowwallet.merged.module@1.8.4/org.jdbi.v3.sqlobject.internal.SqlObjectInitData$1.invoke(Unknown Source)
	at com.sparrowwallet.merged.module@1.8.4/org.jdbi.v3.sqlobject.SqlObjectFactory.lambda$attach$2(Unknown Source)
	at jdk.proxy2/jdk.proxy2.$Proxy29.addWallet(Unknown Source)
	at com.sparrowwallet.sparrow@1.8.4/com.sparrowwallet.sparrow.io.db.DbPersistence.lambda$cleanAndAddWallet$8(Unknown Source)
	at com.sparrowwallet.merged.module@1.8.4/org.jdbi.v3.core.HandleConsumer.lambda$asCallback$0(Unknown Source)
	at com.sparrowwallet.merged.module@1.8.4/org.jdbi.v3.core.Jdbi.withHandle(Unknown Source)
	at com.sparrowwallet.merged.module@1.8.4/org.jdbi.v3.core.Jdbi.useHandle(Unknown Source)
	at com.sparrowwallet.sparrow@1.8.4/com.sparrowwallet.sparrow.io.db.DbPersistence.cleanAndAddWallet(Unknown Source)
	at com.sparrowwallet.sparrow@1.8.4/com.sparrowwallet.sparrow.io.db.DbPersistence.storeWallet(Unknown Source)
	at com.sparrowwallet.sparrow@1.8.4/com.sparrowwallet.sparrow.io.Storage.saveWallet(Unknown Source)
	at com.sparrowwallet.sparrow@1.8.4/com.sparrowwallet.sparrow.io.Storage.migrateType(Unknown Source)
	at com.sparrowwallet.sparrow@1.8.4/com.sparrowwallet.sparrow.io.Storage.migrateToDb(Unknown Source)
	at com.sparrowwallet.sparrow@1.8.4/com.sparrowwallet.sparrow.io.Storage.loadUnencryptedWallet(Unknown Source)
	at com.sparrowwallet.sparrow@1.8.4/com.sparrowwallet.sparrow.io.Storage$LoadWalletService$1.call(Unknown Source)
	at com.sparrowwallet.sparrow@1.8.4/com.sparrowwallet.sparrow.io.Storage$LoadWalletService$1.call(Unknown Source)
	at javafx.graphics@18/javafx.concurrent.Task$TaskCallable.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at javafx.graphics@18/javafx.concurrent.Service.lambda$executeTask$6(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at javafx.graphics@18/javafx.concurrent.Service.lambda$executeTask$7(Unknown Source)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)

**Comment by bry-med** [comment]
After analyzing the text in each JSON file, I see that the wallet I was able to open has a m/84'/0' derivation path but the wallet I'm having trouble with is a legacy wallet with m/44'/0'. 

When I attempt to "Import Wallet" and I select Specter DIY (was running Specter via Umbrel, I never used the desktop app until I updated Specter on my Umbrel; this caused issues thus I decided to try Sparrow). I choose "Legacy", but nothing happens, nothing imports or loads it just sits there.

Not super techy, so starting to get worried here. My Coldcard did not open the legacy wallet when I imported it to Sparrow.
